app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Validator.scala

```diff
@@ -44,42 +41,42 @@ case object Validator {
 
   }
 
-  implicit val format = OFormat(reads, writes)
+  implicit val format: OFormat[Validator] = OFormat(reads, writes)
 
 }
 
 case class HMRCUTRPostcodeCheckValidator(errorMessage: String, utr: FormCtx, postcode: FormCtx) extends Validator
 
 object HMRCUTRPostcodeCheckValidator {
-  private val basic: OFormat[HMRCUTRPostcodeCheckValidator] = Json.format[HMRCUTRPostcodeCheckValidator]
-  private val writesCustom: OWrites[HMRCUTRPostcodeCheckValidator] = OWrites { o =>
+  private val basic: OFormat[HMRCUTRPostcodeCheckValidator] = derived.oformat
+  private val writes: OWrites[HMRCUTRPostcodeCheckValidator] = OWrites { o =>
```

Comment:
I'd get rid of this custom `writes` and used `basic` in its place. To be able to do that we need do appropriate changes on `Validator` reads and writes. It can look like this:
```
case object Validator {
  private val basic: OFormat[Validator] = derived.oformat

  private val templateReads: Reads[Validator] = Reads { json =>
    (json \ "validatorName").as[String] match {
      case "hmrcUTRPostcodeCheck"    => json.validate[HMRCUTRPostcodeCheckValidator]
      case "bankAccountModulusCheck" => json.validate[BankAccoutnModulusCheck]
      case unsupported               => JsError("Unsupported '" + unsupported + "' kind of validator.")
    }
  }

  private val reads = (basic: Reads[Validator]) | templateReads

  implicit val format: OFormat[Validator] = OFormat(reads, basic)
}
```